### PR TITLE
feat: efficient metadata provider draft

### PR DESCRIPTION
**Pull request by arr00**
Link T-3200

### CODE REVIEW

File: contracts/renderers/BasicMetadataProvider.sol [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+        bytes collectionExternalURL;
+        address royaltyReceiver;
+        uint256 royaltyAmount;
+        PartyNFTRenderer.RenderingMethod renderingMethod;
+    }
+
+    /// @notice Set the metadata for a Party instance.
+    /// @param instance The address of the instance.
+    /// @param metadata The encoded metadata.
+    function setMetadata(address instance, bytes memory metadata) external override {
```
Comment (Craigson) [inline contracts/renderers/BasicMetadataProvider.sol:68]: would `calldata` over `memory` help further reduce gas?

Comment (arr00) [reply contracts/renderers/BasicMetadataProvider.sol:68]: yes good call

**Comment by arr00** [comment]
Gas improvement: on the metadata in the test file, party creation cost decreases from 976135 to 567844, a decrease by 42%.

**Comment by arr00** [comment]
Closed in favor of #283

**Pull request closed without merging**